src/main/java/teammates/ui/template/CourseEditSectionRow.java

```diff
@@ -0,0 +1,196 @@
+package teammates.ui.template;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Const;
+
+public class CourseEditSectionRow {
+    private String sectionName;
+    private InstructorAttributes instructor;
+    private List<ElementTag> permissionInputGroup2;
+    private List<ElementTag> permissionInputGroup3;
+    private ElementTag toggleSessionLevelInSectionButton;
+    private List<CourseEditFeedbackSessionRow> feedbackSessions;
+    private List<List<ElementTag>> specialSections;
+    
+    public CourseEditSectionRow(String sectionName, List<String> sectionNames, int sectionIndex, 
+                                InstructorAttributes instructor, int instructorIndex, 
+                                List<String> feedbackNames) {
+        this.sectionName = sectionName;
+        this.instructor = instructor;
+        feedbackSessions = new ArrayList<CourseEditFeedbackSessionRow>();
+        
+        specialSections = createSpecialSectionsForSectionRow(sectionNames, sectionIndex);
+        permissionInputGroup2 = createPermissionInputGroup2ForSectionRow(sectionIndex);
+        permissionInputGroup3 = createPermissionInputGroup3ForSectionRow(sectionIndex);
+        
+        String content = "";
+        String onClick = "";
+        if (!isSessionsInSectionSpecial()) {
+            content = "Give different permissions for sessions in this section";
+            onClick = "showTuneSessionnPermissionsDiv(" + instructorIndex + ", " + sectionIndex + ")";
+        } else {
+            content = "Hide session-level permissions";
+            onClick = "hideTuneSessionnPermissionsDiv(" + instructorIndex + ", " + sectionIndex + ")";
+        }
+        
+        String id = "toggleSessionLevelInSection" + sectionIndex + "ForInstructor" + instructorIndex;
+        toggleSessionLevelInSectionButton = createButton(content, "small col-sm-5", id, "javascript:;", null,
+                                                         onClick, false);
+        
+        for (String feedbackName : feedbackNames) {
+            String[] privileges = {Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS,
+                                   Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS,
+                                   Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS};
```

Comment:
This should be declared outside the `for` loop; no point recreating it each time.